pkg/providers/aws/provider_postgres.go

```diff
@@ -203,6 +202,77 @@ func (p *AWSPostgresProvider) createRDSInstance(ctx context.Context, cr *v1alpha
 		}
 		return nil, "modify instance in progress", nil
 	}
+	// Add Tags to Aws Postgres resources
```

Comment:
I feel this function is getting quite long. Maybe we should look at abstracting the tagging and snapshot logic to its own function.

Reply:
did this in a previous pr as stated above test fail as creating a new session with no credentials.

Reply:
Agree it's quite long , a necessary evil of using the same session for tag snapshot as well, but may be worth a refactor